hal/api/toolchain.h

```diff
@@ -207,6 +207,10 @@
  *  Mark a function declaration as deprecated, if it used then a warning will be
  *  issued by the compiler possibly including the provided message. Note that not
  *  all compilers are able to display the message.
+ *
+ *  @note
```

Comment:
Can do, though we'd have to go with `MBED_DEPRECATED_UNTIL` for the date to remain meaningful in that case. Also it may become less clear what date to put for future patches with deprecations.

I could change the wording to not be a garuntee. Maybe "Functions in the mbed library **should** be marked as deprecated with a minimum expiration of 6 months".

@BlackstoneEngineering, does this seem fine?